src/sentry/static/sentry/app/views/settings/components/dataPrivacyRules/rulesList.tsx

```diff
@@ -1,96 +1,214 @@
+              }
+            : {};
+
+          return (
+            <React.Fragment key={id}>
+              <GridCell {...gridCellProps}>
+                <InnerCell>
+                  <TextOverflow>{`[${methodLabel.label}]`}</TextOverflow>
+                </InnerCell>
+              </GridCell>
+              <GridCell {...gridCellProps}>
+                <InnerCell>
+                  <TextOverflow>{`[${methodDescription}]`}</TextOverflow>
+                </InnerCell>
+              </GridCell>
+              <GridCell {...gridCellProps}>{t('from')}</GridCell>
+              <GridCell {...gridCellProps}>
+                <InnerCell>
+                  <TextOverflow>{`[${source}]`}</TextOverflow>
+                </InnerCell>
+              </GridCell>
+              {onShowEditRuleModal && (
+                <StyledGridCell {...gridCellProps}>
+                  <Button
+                    label={t('Edit Rule')}
+                    size="small"
+                    onClick={onShowEditRuleModal(id)}
+                    icon={<IconEdit />}
+                    disabled={disabled}
+                  />
+                </StyledGridCell>
+              )}
+              {onDeleteRule && (
+                <StyledGridCell {...gridCellProps}>
+                  <Button
+                    label={t('Delete Rule')}
+                    size="small"
+                    onClick={onDeleteRule(id)}
+                    icon={<IconDelete />}
+                    disabled={disabled}
+                  />
+                </StyledGridCell>
+              )}
+            </React.Fragment>
+          );
+        })}
+      </Grid>
+    );
+  }
+}
+
 export default RulesList;
 
-const List = styled('ul')<{isDisabled?: boolean}>`
-  list-style: none;
-  margin: 0;
-  padding: 0;
-  margin-bottom: 0 !important;
+const columnStyle = (p: GridProps & {theme: Theme}) => {
+  if (p.columnQtd === DEFAULT_COLUMN_QUANTITY + 1) {
+    return css`
+      grid-template-columns: max-content auto max-content auto minmax(max-content, 1fr);
+    `;
+  }
+  if (p.columnQtd === DEFAULT_COLUMN_QUANTITY + 2) {
+    return css`
+      grid-template-columns:
+        max-content auto max-content auto minmax(max-content, 1fr)
+        max-content;
+    `;
+  }
+  return css`
+    grid-template-columns: auto minmax(50px, auto) max-content minmax(100px, 1fr);
+  `;
+};
+
+const Grid = styled('div')<GridProps>`
+  display: grid;
+  align-items: center;
+  > *:nth-last-child(-n + ${p => p.columnQtd}) {
+    border-bottom: 0;
+  }
+  > *:nth-child(${p => p.columnQtd}n) {
+    padding-right: ${space(2)};
+  }
+  > *:nth-child(${p => p.columnQtd}n-${p => p.columnQtd - 1}) {
+    padding-left: ${space(2)};
+  }
   ${p =>
     p.isDisabled &&
     `
       color: ${p.theme.gray400};
       background: ${p.theme.gray100};
   `}
+  ${p =>
+    p.hoveredClassname &&
+    `
+    .${p.hoveredClassname} {
+      background-color: ${p.theme.gray100};
+    }
+  `}
```

Comment:
Would this be simpler to implement if the Grid was a table element and the css grid was applied to each row? I've also not see many tables in the application do row highlights on hover.

Reply:
yes, this is a good idea... But sorry... I will prefer to leave as it is now